### PR TITLE
Completing Module 2 exercise

### PR DESCRIPTION
**Pull request by ankanbanerjee22**
------------------
#### React JS Global Learning Program
-------------------
#### MODULE 2 (Exercise) Added tests for the following components,

- Counter
   - Test that component renders initial value provided in props
   - Test that a click event on "decrement" button decrements the displayed value
   - Test that a click event on "increment" button increments the displayed value

- SearchForm
   - Test that component renders an input with the value equal to initial value passed in props
   - Test that after typing to the input and a "click" event on the Submit button, the "onChange" prop is called with proper value
   - Test that after typing to the input and pressing Enter key, the "onChange" prop is called with proper value

- GenreSelect
   - Test that component renders all genres passed in props
   - Test that component highlights a selected genre passed in props
   - Test that after a click event on a genre button component calls "onChange" callback and passes correct genre in arguments

- End-to-End Testing
   - Install and setup [Cypress](https://www.cypress.io/) for end-to-end tests.
   - Covered the logic for one of your components using the framework [Counter component]
   
#### VISUALS
**unit tests passed for all the components** 
image

**testing component flow for counter component using cypress**
e2e-Test-Cypress

**testing e2e flow for counter component using cypress**
e2e_cypress

**E2E executed Steps:**
image

### CODE REVIEW

File: module_1/first-react-app/cypress/support/commands.js
Comment (andreyBalabukha) [inline module_1/first-react-app/cypress/support/commands.js:1]: Do we really need this file?

File: module_1/first-react-app/cypress/support/component-index.html [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  <!-- including materialize css -->
+
+<!-- Compiled and minified CSS -->
+<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/materialize/1.0.0/css/materialize.min.css">
+
+<!-- Compiled and minified JavaScript -->
+<script src="https://cdnjs.cloudflare.com/ajax/libs/materialize/1.0.0/js/materialize.min.js"></script>
+
+<link href="https://fonts.googleapis.com/icon?family=Material+Icons" rel="stylesheet">
+</html>
```
Comment (andreyBalabukha) [inline module_1/first-react-app/cypress/support/component-index.html:22]: Please don't forget to put the blank line in the end. Also please check in other files

**Review by andreyBalabukha — APPROVED** [review]
Well done

File: module_1/first-react-app/cypress/support/component-index.html (hunk 1 above)
Comment (ankanbanerjee22) [reply module_1/first-react-app/cypress/support/component-index.html:22]: have added.

File: module_1/first-react-app/cypress/support/commands.js
Comment (ankanbanerjee22) [reply module_1/first-react-app/cypress/support/commands.js:1]: This was auto generated upon adding cypress, and other files are importing this, so if we need to remove this file then would require to remove those imports as well.

So keeping this for now!

**Pull request merged**